promquery/bulkquerycache.go

```diff
@@ -103,6 +104,10 @@ func (c *BulkQueryCache[K, V]) fillCacheIfNecessary(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("cannot collect %s: %w", q.Description, err)
 		}
+		// prevent empty prometheus results from being processed downstream.
+		if len(vector) == 0 {
+			return errors.New("did not receive any values from prometheus")
```

Comment:
Please include `q.Description` like in the other error nearby, to indicate which query failed.